app/services/session_encryptor.rb

```diff
@@ -8,15 +8,13 @@ def self.build_user_access_key
     build_user_access_key
   end
 
-  # rubocop:disable Security/MarshalLoad
   def self.load(value)
     decrypted = encryptor.decrypt(value, user_access_key)
-    Marshal.load(::Base64.decode64(decrypted))
+    ActiveSupport::JSON.decode(::Base64.decode64(decrypted))
```

Comment:
Also, I think we may need to do a rescue here to catch errors parsing JSON. sessions that already exist while the code is being deployed will be marshal-encoded, only new sessions will be JSON-encoded (this is another +1/-1 issue we'll need to fix over multiple deploys at least 15 minutes apart so old sessions expire)